example/passthrough_hp_uds.cc

```diff
@@ -0,0 +1,1406 @@
+    for (auto opt : mount_options) {
+        string_split(opt, flattened_mount_opts, ",");
+    }
+
+    bool found_fsname = false;
+    for (auto opt : flattened_mount_opts) {
+        if (opt.find("fsname=") == 0) {
+            found_fsname = true;
+            continue;
+        }
+
+        /* Filter out some obviously incorrect options. */
+        if (opt == "fd") {
+            std::cout << argv[0] << ": Unsupported mount option: " << opt << "\n";
+            print_usage(argv[0]);
+            exit(2);
+        }
+    }
+    if (!found_fsname) {
+        flattened_mount_opts.push_back("fsname=" + fs.source);
+    }
+    flattened_mount_opts.push_back("default_permissions");
+    fs.fuse_mount_options = string_join(flattened_mount_opts, ',');
+    return options;
+}
+
+
+static void maximize_fd_limit() {
+    struct rlimit lim {};
+    auto res = getrlimit(RLIMIT_NOFILE, &lim);
+    if (res != 0) {
+        warn("WARNING: getrlimit() failed with");
+        return;
+    }
+    lim.rlim_cur = lim.rlim_max;
+    res = setrlimit(RLIMIT_NOFILE, &lim);
+    if (res != 0)
+        warn("WARNING: setrlimit() failed with");
+}
+
+
+int main(int argc, char *argv[]) {
+
+    struct fuse_loop_config *loop_config = NULL;
+
+    // Parse command line options
+    auto options {parse_options(argc, argv)};
+
+    // We need an fd for every dentry in our the filesystem that the
+    // kernel knows about. This is way more than most processes need,
+    // so try to get rid of any resource softlimit.
+    maximize_fd_limit();
+
+    // Initialize filesystem root
+    fs.root.fd = -1;
+    fs.root.nlookup = 9999;
+    fs.timeout = options.count("nocache") ? 0 : 86400.0;
+
+    struct stat stat;
+    auto ret = lstat(fs.source.c_str(), &stat);
+    if (ret == -1)
+        err(1, "ERROR: failed to stat source (\"%s\")", fs.source.c_str());
+    if (!S_ISDIR(stat.st_mode))
+        errx(1, "ERROR: source is not a directory");
+    fs.src_dev = stat.st_dev;
+
+    fs.root.fd = open(fs.source.c_str(), O_PATH);
+    if (fs.root.fd == -1)
+        err(1, "ERROR: open(\"%s\", O_PATH)", fs.source.c_str());
+
+    // Initialize fuse
+    fuse_args args = FUSE_ARGS_INIT(0, nullptr);
+    if (fuse_opt_add_arg(&args, argv[0]) ||
+        fuse_opt_add_arg(&args, "-o") ||
+        fuse_opt_add_arg(&args, fs.fuse_mount_options.c_str()) ||
+        (fs.debug_fuse && fuse_opt_add_arg(&args, "-odebug")))
+        errx(3, "ERROR: Out of memory");
+
+    fuse_lowlevel_ops sfs_oper {};
+    assign_operations(sfs_oper);
+    auto se = fuse_session_new(&args, &sfs_oper, sizeof(sfs_oper), &fs);
+    if (se == nullptr)
+        goto err_out1;
+
+    if (fuse_set_signal_handlers(se) != 0)
+        goto err_out2;
+
+    // Don't apply umask, use modes exactly as specified
+    umask(0);
+
+    // Mount and run main loop
+    loop_config = fuse_loop_cfg_create();
+
+    if (fs.num_threads != -1)
+        fuse_loop_cfg_set_idle_threads(loop_config, fs.num_threads);
```

Comment:
That is a deprecated option, please use `_max_threads`. Unless you want to get persistent thread creation/destruction overhead. 
Hmm, I see, looks like I had accidentally also used "_idle_threads" in passthrough_hp.cc - needs to be fixed.

Reply:
Note that the first commit simply copies an existing example.

Reply:
We don't need to propagate that error even further. As I wrote, I would prefer a common file (feel free to copy it for that) so that we avoid unnecessary code dup.
I will submit a patch later on to fix the existing _idle_threads issue.

Reply:
This might be a good idea.
I will split the logic as described and try to keep changes simple as they are still examples.